cellprofiler/pipeline.py

```diff
@@ -1117,6 +1117,87 @@ def encode_txt(self, s):
         s = s.replace('[', '\\x5B').replace(']', '\\x5D')
         return s
 
+    def to_dictionary(self):
+        pipeline_dictionary = {
+            "cookie": COOKIE,
+            "modules": [],
+            "version": cellprofiler.__version__
+        }
+
+        for module in self.__modules:
+            module_dictionary = {
+                "batch_state": str(getattr(module, "batch_state")),
+                "displayed": getattr(module, "show_window"),
+                "enabled": getattr(module, "enabled"),
+                "name": module.module_name,
+                "notes": getattr(module, "notes"),
+                "paused": getattr(module, "wants_pause"),
+                "settings": {},
+                "version": getattr(module, "variable_revision_number")
+            }
+
+            for setting in module.settings():
+                # FIXME: Why does CellProfiler have hidden settings?
+                if not setting.text:
+                    continue
+
+                v = self.encode_txt(setting.unicode_value.encode('utf-8'))
+
+                module_dictionary["settings"][setting.text] = v
+
+            pipeline_dictionary["modules"] += [module_dictionary]
+
+        return pipeline_dictionary
+
+    def export_json(self, pathname):
+        """
+
+        """
+        dictionary = self.to_dictionary()
+
+        with open(pathname, "w") as stream:
```

Comment:
RE above, this can be a `codecs.open` with `utf-8`, then we don't have to encode the variables and they can still be readable!